backend/src/main/java/com/woowacourse/kkogkkog/infrastructure/SlackRequester.java

```diff
@@ -0,0 +1,102 @@
+    private static final String CLIENT_ID_PARAMETER_NAME = "client_id";
+    private static final String SECRET_ID_PARAMETER_NAME = "client_secret";
+
+    private final String clientId;
+    private final String secretId;
+    private final WebClient oAuthLoginClient;
+    private final WebClient userClient;
+
+    public SlackRequester(
+        @Value("${security.slack.client-id}") String clientId,
+        @Value("${security.slack.secret-id}") String secretId,
+        @Value(OAUTH_LOGIN_URI) String oAuthLoginUri,
+        @Value(OAUTH_USER_INFO) String userInfoUri,
+        WebClient webClient) {
+        this.clientId = clientId;
+        this.secretId = secretId;
+        this.oAuthLoginClient = toWebClient(webClient, oAuthLoginUri);
+        this.userClient = toWebClient(webClient, userInfoUri);
+    }
+
+    private WebClient toWebClient(WebClient webClient, String baseUrl) {
+        return webClient.mutate()
+            .baseUrl(baseUrl)
+            .defaultHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
+            .build();
+    }
+
+    public SlackUserInfo getUserInfoByCode(final String code) {
+        String token = requestAccessToken(code);
+        return requestUserInfo(token);
+    }
+
+    private String requestAccessToken(String code) {
+        Map<String, Object> responseBody = oAuthLoginClient
+            .post()
+            .uri(uriBuilder -> toRequestTokenUri(code, uriBuilder))
+            .headers(this::setHeaders)
+            .retrieve()
+            .bodyToMono(PARAMETERIZED_TYPE_REFERENCE)
+            .blockOptional()
+            .orElseThrow(UnableToGetTokenResponseException::new);
+        validateResponseBody(responseBody);
+
+        return responseBody.get("access_token").toString();
+    }
+
+    private URI toRequestTokenUri(String code, UriBuilder uriBuilder) {
+        return uriBuilder
+            .queryParam(CODE_PARAMETER_NAME, code)
+            .queryParam(CLIENT_ID_PARAMETER_NAME, clientId)
+            .queryParam(SECRET_ID_PARAMETER_NAME, secretId)
+            .build();
+    }
+
+    private void setHeaders(HttpHeaders header) {
+        header.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
+        header.setAcceptCharset(Collections.singletonList(StandardCharsets.UTF_8));
+    }
+
+    private void validateResponseBody(Map<String, Object> responseBody) {
+        if (!responseBody.containsKey("access_token")) {
+            throw new ErrorResponseToGetAccessTokenException(responseBody.get("error").toString());
+        }
+    }
+
+    private SlackUserInfo requestUserInfo(String token) {
+        SlackUserInfo slackUserInfoResponse = userClient
+            .get()
+            .headers(httpHeaders -> httpHeaders.setBearerAuth(token))
+            .retrieve()
+            .bodyToMono(SlackUserInfo.class)
+            .blockOptional()
+            .orElseThrow(UnableToGetUserInfoResponseException::new);
+
```

Comment:
```suggestion
    private SlackUserInfo requestUserInfo(String token) {
        SlackUserInfo slackUserInfoResponse = userClient
            .get()
            .headers(httpHeaders -> httpHeaders.setBearerAuth(token))
            .retrieve()
            .bodyToMono(SlackUserInfo.class)
            .blockOptional()
            .orElseThrow(UnableToGetUserInfoException::new);
```

UnableToGetUserInfoResponseExcepion 과 UnauthencatedTokenException 모두 유효하지 않은 토큰입니다 라는 문구를 반환합니다. 구분하신 이유가 있을까요?

Exception이름이 너무 길어져 `Response` 부분을 빼도 좋을 것 같아요